Pocket/pocket-ios
# Native Collections: cache stories in Core Data

## Summary
* This PR uses Core Data to cache collection stories, and uses cached data (when available) to display collections in the native collections view

## References 
* branch name

## Implementation Details
* Update `CollectionViewModel`, `CollectionViewController`, use a `RichFetchedResultController` to cache collection stories and update the `UICollectionViewDiffableDataSource` accordingly

## Test Steps
* Tap Home, then scroll until you find a collection or
* Tap Home, enter a slate detail and scroll until you find a collection
* Tap the collection cell:
    - if it's the first time you access it, expect some loading time for the collection to load remotely
    - Navigate back, then re-select the same collection
    - make sure the collection loads rapidly

## PR Checklist:

## Screenshots



https://github.com/Pocket/pocket-ios/assets/34376330/320aeb2d-1b0c-422d-9e48-e638375e7382
